openai_adapter.py

```
@@ -0,0 +1,370 @@
+        result = ctx.mcp_server.tools[tool_name](ctx, **params)
+        return {"result": result}
+    except Exception as e:
+        # Structured error response
+        raise ToolExecutionError(tool_name, str(e), {
+            "exception_type": type(e).__name__,
+            "traceback": traceback.format_exc()
+        })
+
+def main():
+    """Legacy entry point for backward compatibility."""
+    warnings.warn(
```

Ruff (0.8.2)

No explicit `stacklevel` keyword argument found

(B028)